tokio/src/runtime/builder.rs

```diff
@@ -800,6 +803,31 @@ impl Builder {
         }
     }
 
+    /// Creates the configured `LocalRuntime`.
+    ///
+    /// The returned `LocalRuntime` instance is ready to spawn tasks.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::runtime::Builder;
+    ///
+    /// let rt  = Builder::new_current_thread().build_local(&mut Default::default()).unwrap();
+    ///
+    /// rt.block_on(async {
+    ///     println!("Hello from the Tokio runtime");
+    /// });
+    /// ```
+    #[allow(unused_variables, unreachable_patterns)]
+    #[cfg(tokio_unstable)]
+    #[cfg_attr(docsrs, doc(cfg(tokio_unstable)))]
+    pub fn build_local(&mut self, options: &mut LocalOptions) -> io::Result<LocalRuntime> {
```

Comment:
I think it's weird to take a `&mut LocalOptions` here. Why not `LocalOptions` by-value?

Reply:
I decided to match the semantics of the builder itself, which uses a mutable reference to build.

I lean towards taking a reference of some sort rather than passing by-value, but I don't think it needs to be a mutable one necessarily.

Is there an advantage you can see in passing the options by-value instead of by reference?

Reply:
With a mutable reference, the caller has to write `&mut` to pass their argument. This isn't an issue for the self parameter as it just uses the dot operator.

Reply:
Done.